app/scripts/components/common/mapbox/layers/raster-timeseries.tsx

```diff
@@ -470,10 +471,18 @@ export function MapLayerRasterTimeseries(props: MapLayerRasterTimeseriesProps) {
     if (!stacCollection.length) return;
     const layerBounds = getMergedBBox(stacCollection);
 
-    if (checkFitBoundsFromLayer(layerBounds, mapInstance)) {
-      mapInstance.fitBounds(layerBounds, { padding: FIT_BOUNDS_PADDING });
+    // Prefer layer defined bounds to STAC collection bounds.
+    const usableBounds = (bounds?.length === 4 ? bounds : layerBounds) as [
+      number,
+      number,
+      number,
+      number
+    ];
+
+    if (checkFitBoundsFromLayer(usableBounds, mapInstance)) {
+      mapInstance.fitBounds(usableBounds, { padding: FIT_BOUNDS_PADDING });
```

Comment:
This logic is the same as what you added in `vector-timeseries`. 
Might be worth exploring running it at the map level (maybe in `mapbox/index.tsx`), or as an alternative provide a hook to reduce reprition.

Reply:
Abstracted the function to a hook. There should be less repetition now, but I had to get the stac bbox for the vector data as well.